### PR TITLE
Partially annotate Vavr APIs to improve IndexChecker guarantees

### PR DESCRIPTION
**Pull request by PawelLipski**

### CODE REVIEW

File: branchLayoutImpl/src/main/java/com/virtuslab/branchlayout/impl/BranchLayoutFileParser.java [hunk 1]
```diff
@@ -35,13 +35,14 @@ public BranchLayout parse() throws BranchLayoutException {
...
   /**
-   * Searches for first line preceded some indent character and based on that sets {@code indentCharacter} and
+   * Searches for first line starting with an indent character and based on that sets {@code indentCharacter} and
    * {@code levelWidth} fields.
    */
-  @SuppressWarnings("index:argument.type.incompatible")
   private void deriveIndentCharacter(List<String> lines) {
-    var firstLineWithBlankPrefixOption = lines.find(line -> line.startsWith(" ") || line.startsWith("\t"));
-    if (firstLineWithBlankPrefixOption.isDefined()) {
+    var firstLineWithBlankPrefixOption = lines.find(line -> line.startsWith(" ") || line.startsWith("\t"))
```
Comment (mkondratek) [inline branchLayoutImpl/src/main/java/com/virtuslab/branchlayout/impl/BranchLayoutFileParser.java:42]: s/Option/Optional

Comment (PawelLipski) [reply branchLayoutImpl/src/main/java/com/virtuslab/branchlayout/impl/BranchLayoutFileParser.java:42]: Done on a follow-up PR

File: frontendGraph/src/main/java/com/virtuslab/gitmachete/frontend/graph/repository/RepositoryGraphBuilder.java [hunk 2]
```diff
@@ -1,9 +1,9 @@
 package com.virtuslab.gitmachete.frontend.graph.repository;
 
 import java.util.ArrayList;
-import java.util.List;
 import java.util.Optional;
 
+import io.vavr.collection.List;
```
Comment (mkondratek) [inline frontendGraph/src/main/java/com/virtuslab/gitmachete/frontend/graph/repository/RepositoryGraphBuilder.java:6]: **line 36:** `public static final IBranchGetCommitsStrategy EMPTY_GET_COMMITS = __ -> io.vavr.collection.List.empty();`

We want just `List.empty()` here, don't we?

Comment (PawelLipski) [reply frontendGraph/src/main/java/com/virtuslab/gitmachete/frontend/graph/repository/RepositoryGraphBuilder.java:6]: Fixed on #182

**Pull request merged**